lib/_http_outgoing.js

```diff
@@ -426,6 +430,26 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns somewhat shallow copy of current outgoing headers. Array header
+// values will be cloned but no deeper cloning will occur.
+OutgoingMessage.prototype.getHeaders = function getHeaders(namesOnly) {
+  const headers = this._headers;
+  const keys = Object.keys(headers);
+  if (namesOnly)
+    return keys;
+  const ret = new OutgoingHeaders();
+  for (var i = 0; i < keys.length; ++i) {
+    const key = keys[i];
+    const val = headers[key][1];
```

Comment:
I didn't specifically compare vs. `var`, but it's not aborting optimizations because of it or anything (which can still in happen in some cases in V8 5.4).